metrics/internal/collectors/storageconsumer.go

```diff
@@ -59,12 +114,84 @@ func (c *StorageConsumerCollector) Run(stopCh <-chan struct{}) {
 	go c.Informer.Run(stopCh)
 }
 
+func (c *StorageConsumerCollector) getClusterVersion() string {
+
+	clusterVersion, err := c.verClient.ClusterVersions().Get(context.TODO(), "version", metav1.GetOptions{})
+	if err != nil {
+		klog.Errorf("Failed to get clusterVersion: %v", err)
+	}
+
+	var version string
+	for idx := range clusterVersion.Status.History {
+		candidate := &clusterVersion.Status.History[idx]
+		if candidate.State == configv1.CompletedUpdate {
+			version = candidate.Version
+			break
+		}
+	}
+
+	if version == "" {
+		klog.Errorf("Unable to find ocp version with completed update")
+	}
+
+	return version
+}
+
+// encodes version padding with 3 zeros for each part making suitable
+// for numerical comparisions
+// ex: 4.10.3 -> 004 010 003 -> 4010003
+func encodeVersion(v string) int {
+	if v == "" {
+		return -1
+	}
+
+	parts := strings.Split(v, ".")
+	if len(parts) != 3 {
+		return -1
+	}
+	sb := make([]string, 3)
+	for i := range sb {
+		sb[i] = fmt.Sprintf("%03s", parts[i])
+	}
+
+	ver := strings.Join(sb, "")
+	encode, err := strconv.Atoi(ver)
+	if err != nil {
+		return -1
+	}
+
+	return encode
+}
+
 func (c *StorageConsumerCollector) collectStorageConsumersMetadata(storageConsumers []*ocsv1alpha1.StorageConsumer, ch chan<- prometheus.Metric) {
+
+	ch <- prometheus.MustNewConstMetric(c.ProviderClusterVersion,
+		prometheus.GaugeValue, float64(encodeVersion(c.getClusterVersion())),
+	)
+
+	ch <- prometheus.MustNewConstMetric(c.ProviderOperatorVersion,
+		prometheus.GaugeValue, float64(encodeVersion(version.GetVersion())),
+	)
+
 	for _, storageConsumer := range storageConsumers {
 		ch <- prometheus.MustNewConstMetric(c.StorageConsumerMetadata,
 			prometheus.GaugeValue, 1,
 			storageConsumer.Name,
 			string(storageConsumer.Status.State))
+
+		ch <- prometheus.MustNewConstMetric(c.LastHeartbeat,
+			prometheus.GaugeValue, float64(storageConsumer.Status.LastHeartbeat.Time.Unix()),
```

Comment:
Probably a heartbeat based on a timestamp would be of type "counter" instead "gauge". Do you expect to have decrements on this metric?

Reply:
There'll be no decrements, however _count and _sum derived metrics from a counter type doesn't mean anything in this context and so I used Gauge to represent the instant value with no correlation to previous values.